core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -152,28 +168,49 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
     }
 
+  def enumerateSnapshotsByWorkspaceName(workspaceName: WorkspaceName,
+                                        offset: Int,
+                                        limit: Int,
+                                        referencedSnapshotId: Option[UUID] = None
+  ): Future[SnapshotListResponse] =
+    getV2WorkspaceContextAndPermissions(workspaceName,
+                                        SamWorkspaceActions.read,
+                                        Some(WorkspaceAttributeSpecs(all = false))
+    ).flatMap(workspaceContext =>
+      enumerateSnapshots(workspaceContext.workspaceIdAsUUID, offset, limit, referencedSnapshotId)
+    )
+
+  def enumerateSnapshotsById(workspaceId: String,
+                             offset: Int,
+                             limit: Int,
+                             referencedSnapshotId: Option[UUID] = None
+  ): Future[SnapshotListResponse] =
+    getV2WorkspaceContextAndPermissionsById(workspaceId,
+                                            SamWorkspaceActions.read,
+                                            Some(WorkspaceAttributeSpecs(all = false))
+    ).flatMap(workspaceContext =>
+      enumerateSnapshots(workspaceContext.workspaceIdAsUUID, offset, limit, referencedSnapshotId)
+    )
+
   /**
     * return a given page of snapshot references from Workspace Manager, optionally returning only those
     * snapshot references that refer to a supplied TDR snapshotId.
     *
-    * @param workspaceName the workspace owning the snapshot references
+    * @param workspaceId the id of the workspace owning the snapshot references
     * @param offset pagination offset for the list
     * @param limit pagination limit for the list
     * @param referencedSnapshotId the TDR snapshotId for which to return matching references
     * @return the list of snapshot references
     */
-  def enumerateSnapshots(workspaceName: WorkspaceName,
-                         offset: Int,
-                         limit: Int,
-                         referencedSnapshotId: Option[UUID] = None
+  private def enumerateSnapshots(workspaceId: UUID,
+                                 offset: Int,
+                                 limit: Int,
+                                 referencedSnapshotId: Option[UUID] = None
   ): Future[SnapshotListResponse] =
-    getV2WorkspaceContextAndPermissions(workspaceName,
-                                        SamWorkspaceActions.read,
-                                        Some(WorkspaceAttributeSpecs(all = false))
-    ).map { workspaceContext =>
+    Future.successful {
       referencedSnapshotId match {
-        case None     => retrieveSnapshotReferences(workspaceContext.workspaceIdAsUUID, offset, limit)
-        case Some(id) => findBySnapshotId(workspaceContext.workspaceIdAsUUID, id, offset, limit)
+        case None     => retrieveSnapshotReferences(workspaceId, offset, limit)
+        case Some(id) => findBySnapshotId(workspaceId, id, offset, limit)
       }
     }
```

Comment:
this unnecessarily returns a `Future`. The return value is set to be `Future[SnapshotListResponse]`. Internally to the method, it only calls synchronous operations but then wraps the return value inside `Future.successful(…)`. This could be:
```scala
  private def enumerateSnapshots(workspaceId: UUID,
                                 offset: Int,
                                 limit: Int,
                                 referencedSnapshotId: Option[UUID] = None
  ): SnapshotListResponse =
    referencedSnapshotId match {
      case None     => retrieveSnapshotReferences(workspaceId, offset, limit)
      case Some(id) => findBySnapshotId(workspaceId, id, offset, limit)
    }
```

Reply:
Ah yes that was an earlier iteration when I didn't have a separate `enumerateSnapshotsById` method and was trying to wrangle correct return values... thank you!